lib/xcode.js

```diff
@@ -198,6 +199,36 @@ const getMaxIOSSDK = _.memoize(
   }
 );
 
+function searchForConnectedDevices(obj, acc) {
+  if (obj.constructor === Array) {
+    obj.forEach( function(item) {
+      searchForConnectedDevices(item, acc);
+    });
+  }else {
+    if (obj._name && obj._name.substring(0,4) === "iPad" ||
+      obj._name && obj._name.substring(0,6) === "iPhone") {
+      acc.push(obj);
+    } else if (obj._items) {
+      searchForConnectedDevices(obj._items, acc);
+    }
+  }
+}
+
+async function getConnectedDevicesWithoutRetry () {
+
+  let [stdout] = await exec('/usr/sbin/system_profiler -xml SPUSBDataType', {maxBuffer: 524288, timeout: XCODE_SELECT_TIMEOUT});
+  var val = plist.parse(stdout);
+  var devices = [];
+  searchForConnectedDevices(val[0]._items, devices);
+  return devices;
+}
+
+const getConnectedDevices = _.memoize(
```

Comment:
@penguinho are you sure you want to memoize this function? Devices connected could change during an appium run and we would want to detect that.
Or are you confident that it shouldn't change (or does this function hurt performance)

If you do want it to be memoized, can you add it to the list of functions in `clearInternalCache`?